nixos/modules/services/networking/hostapd.nix

```diff
@@ -1,222 +1,1011 @@
+      ''}
+
+      ${optionalString ifcfg.wifi5.enable ''
+        ##### IEEE 802.11ac (WiFi 5) related configuration #####################################
+
+        ieee80211ac=1
+        vht_capab=${concatMapStrings (x: "[${x}]") ifcfg.wifi5.capabilities}
+        require_vht=${bool01 ifcfg.wifi5.require}
+        vht_oper_chwidth=${ifcfg.wifi5.operatingChannelWidth}
+      ''}
+      ${optionalString ifcfg.wifi6.enable ''
+        ##### IEEE 802.11ax (WiFi 6) related configuration #####################################
+
+        ieee80211ax=1
+        require_he=${bool01 ifcfg.wifi6.require}
+        he_oper_chwidth=${ifcfg.wifi6.operatingChannelWidth}
+        he_su_beamformer=${bool01 ifcfg.wifi6.singleUserBeamformer}
+        he_su_beamformee=${bool01 ifcfg.wifi6.singleUserBeamformee}
+        he_mu_beamformer=${bool01 ifcfg.wifi6.multiUserBeamformer}
+      ''}
+      ${optionalString ifcfg.wifi7.enable ''
+        ##### IEEE 802.11be (WiFi 7) related configuration #####################################
+
+        ieee80211be=1
+        eht_oper_chwidth=${ifcfg.wifi7.operatingChannelWidth}
+        eht_su_beamformer=${bool01 ifcfg.wifi7.singleUserBeamformer}
+        eht_su_beamformee=${bool01 ifcfg.wifi7.singleUserBeamformee}
+        eht_mu_beamformer=${bool01 ifcfg.wifi7.multiUserBeamformer}
+      ''}
+
+      ##### WPA/IEEE 802.11i configuration ##########################################
+
+      # Encrypt management frames to protect against deauthentication and similar attacks
+      ieee80211w=${ifcfg.managementFrameProtection}
+      ${optionalString (ifcfg.authentication.mode == "none") ''
+        wpa=0
+      ''}
+      ${optionalString (ifcfg.authentication.mode == "wpa3-sae") ''
+        wpa=2
+        wpa_key_mgmt=SAE
+        # Derive PWE using both hunting-and-pecking loop and hash-to-element
+        sae_pwe=2
+        # Prevent downgrade attacks by indicating to clients that they should
+        # disable any transition modes from now on.
+        transition_disable=0x01
+      ''}
+      ${optionalString (ifcfg.authentication.mode == "wpa3-sae-transition") ''
+        wpa=2
+        wpa_key_mgmt=WPA-PSK-SHA256 SAE
+      ''}
+      ${optionalString (ifcfg.authentication.mode == "wpa2-sha256") ''
+        wpa=2
+        wpa_key_mgmt=WPA-PSK-SHA256
+      ''}
+      ${optionalString (ifcfg.authentication.mode != "none") ''
+        wpa_pairwise=${concatStringsSep " " ifcfg.authentication.pairwiseCiphers}
+        rsn_pairwise=${concatStringsSep " " ifcfg.authentication.pairwiseCiphers}
+      ''}
+
+      ${optionalString (ifcfg.authentication.wpaPassword != null) ''
+        wpa_passphrase=${ifcfg.authentication.wpaPassword}
+      ''}
+      ${optionalString (ifcfg.authentication.wpaPskFile != null) ''
+        wpa_passphrase=${ifcfg.authentication.wpaPskFile}
+      ''}
+      ${optionalString (ifcfg.authentication.saePasswords != []) (concatMapStrings (pw: "sae_password=${pw}\n") ifcfg.authentication.saePasswords)}
```

Comment:
this cannot work as `pw` is not necessarily a string :P

Reply:
damn I missed that when promoting saePasswords to a set, thanks. Will push an update in a minute.